### PR TITLE
SSCS-10294 Support for adjournment feature flag

### PR DESCRIPTION
**Pull request by mcareyma**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/SSCS-10294

### Change description ###
Added support for adjournment feature flag, so that old code is run without the flag and the new code
only runs with the flag, and the tests test both.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsDurationMapping.java [hunk 1]
```diff
@@ -19,54 +19,93 @@
...
+            if (DURATION_UNITS_SESSIONS.equalsIgnoreCase(caseData.getAdjournCaseNextHearingListingDurationUnits())) {
+                return Integer.parseInt(adjournCaseNextHearingListingDuration) * DURATION_SESSIONS_MULTIPLIER;
+            }
+            if (DURATION_UNIT_HOURS.equalsIgnoreCase(caseData.getAdjournCaseNextHearingListingDurationUnits())) {
+                // TODO Adjournments - check no other measurement than hours, sessions and null
+                return Integer.parseInt(adjournCaseNextHearingListingDuration) * DURATION_HOURS_MULTIPLIER;
+            }
+        }
+
+        return null;
```
Comment (DGordonV1) [inline src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsDurationMapping.java:79]: No change, just wondering if returning null has any undesired impact here? I note we're already returning null in other methods.

Also, any reason to remove if/else if for multiple if statements? I'm not fussed on which is being used when returning within them, as you're not using additional overhead for the other ifs when a condition is met, so all good, just minor comment really.

File: src/test/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsDurationMappingTest.java [hunk 2]
```diff
@@ -241,10 +244,17 @@ void getHearingDurationBenefitIssueCodesPaper() {
...
     void getHearingDurationBenefitIssueCodesFaceToFace() {
+        if (!referenceDataServiceHolder.isAdjournmentFlagEnabled()) {
+            given(referenceDataServiceHolder.isAdjournmentFlagEnabled()).willReturn(true);
+        }
+        if (!referenceDataServiceHolder.isAdjournmentFlagEnabled()) {
+            given(referenceDataServiceHolder.isAdjournmentFlagEnabled()).willReturn(true);
+        }
+        if (!referenceDataServiceHolder.isAdjournmentFlagEnabled()) {
+            given(referenceDataServiceHolder.isAdjournmentFlagEnabled()).willReturn(true);
+        }
```
Comment (DGordonV1) [inline src/test/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsDurationMappingTest.java:255]: Repeated three times.

**Review by DGordonV1 — CHANGES_REQUESTED** [review]

File: src/test/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsDurationMappingTest.java (hunk 2 above)
Comment (amoslv1) [reply src/test/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsDurationMappingTest.java:255]: done

**Review by DGordonV1 — DISMISSED** [review]

File: src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsDurationMapping.java [hunk 3]
```diff
@@ -19,54 +19,93 @@
...
         }
 
         return nonNull(duration) ? duration : DURATION_DEFAULT;
     }
 
-    public static Integer getHearingDurationAdjournment(SscsCaseData caseData) {
+    public static Integer getHearingDurationAdjournment(
+        SscsCaseData caseData,
+        ReferenceDataServiceHolder referenceDataServiceHolder
+    ) {
```
Comment (dattipoeHMCTS) [inline src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsDurationMapping.java:58]: (m) weird formatting

File: src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsDurationMapping.java [hunk 4]
```diff
@@ -19,54 +19,93 @@
...
+        return referenceDataServiceHolder.isAdjournmentFlagEnabled() // TODO remove when flag enabled
+            ? getHearingDurationAdjournmentNew(caseData)
+            : getHearingDurationAdjournmentOld(caseData);
+    }
 
-        if (DURATION_TYPE_NON_STANDARD_TIME_SLOT.equalsIgnoreCase(caseData.getAdjournCaseNextHearingListingDurationType())
-            && isNotBlank(caseData.getAdjournCaseNextHearingListingDuration())) {
-            int duration = Integer.parseInt(caseData.getAdjournCaseNextHearingListingDuration());
+    private static Integer getHearingDurationAdjournmentOld(SscsCaseData caseData) {
```
Comment (dattipoeHMCTS) [inline src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsDurationMapping.java:64]: What do 'old' and 'new' mean?

**Comment by amoslv1** [comment]
Changes being made on another branch, so closing this PR

**Pull request closed without merging**